host_source.go

```diff
@@ -174,6 +174,21 @@ func (h *HostInfo) connectAddressLocked() (net.IP, string) {
 	return net.IPv4zero, "invalid"
 }
 
+// NativeAddress returns address broadcasted between node to nodes.
+// It's either `broadcast_address` if host info is read from system.local or `peer` if read from system.peers.
+// This address is also used in CQL Event emitted on topology/status changes.
+func (h *HostInfo) NativeAddress() net.IP {
```

Comment:
The type emitted in topology/status change events is `[inet]`:

https://github.com/apache/cassandra/blob/39e89fd636ee4343eb2201820da87881cbc749e2/doc/native_protocol_v4.spec#L760-L770

`[inet]` is IP address and port:

https://github.com/apache/cassandra/blob/39e89fd636ee4343eb2201820da87881cbc749e2/doc/native_protocol_v4.spec#L241-L245

It seems we actually don't get the port from `system.local` and `system.peers` though as the `broadcast_address` and `peer` columns are `inet` CQL data type (IP address only), which is distinct from the `[inet]`. See also https://github.com/gocql/gocql/issues/1634#issuecomment-1167002493

I don't know if we can do something else than ignore the port though as the port is not in system.peers table. We can at least clarify the doc comment a little bit:

```
// This IP address is also part of CQL Event emitted on topology/status changes,
// but does not uniquely identify the node in case multiple nodes use the same IP address.
```

Reply:
Both Cassandra and Scylla returns keep IP address in `system.local` and `system.peers`. Extended the documentation with your suggestion.

Reply:
It seems that we can get the port from `system.peers_v2` for Cassandra 4.0 and up (see https://github.com/gocql/gocql/issues/1634#issuecomment-1167091619), so `NativeAddress()` should probably return both IP address and port. If we had NativeAddress() and NativePort() as separate methods, we would have to duplicate the selection logic.

Also there are `native_address` and `native_port` fields in `system.peers_v2`. Should NativeAddress() use those for Cassandra 4.0 or are those unrelated and we should rename the NativeAddress() method not to clash?

Reply:
That table doesn't exist for Yugabyte so it would need to check the error and fallback if we end up trying to use that table: https://github.com/yugabyte/yugabyte-db/commit/fc5e713354c388391b01170871d4e65c8623f69f

Reply:
Currently https://github.com/gocql/gocql/pull/1635 has check for server version >= 4.0.0 to determine whether to use system.peers_v2. Would that check work for Yugabyte or should we also add the fallback in case the table does not exist? For Scylla it should work with the version check as it returns server version lower than 4 currently.

Reply:
Changed name to NodeToNodeAddress() to avoid clash with how columns are called in `system.peers_v2`.

Reply:
sure, fixed